soco/core.py

```diff
@@ -1255,6 +1266,35 @@ def get_sonos_playlists(self, *args, **kwargs):
         args = tuple(['sonos_playlists'] + list(args))
         return self.get_music_library_information(*args, **kwargs)
 
+    def get_sonos_playlist_by_attr(self, attr_name, match):
+        """ Return the first Sonos Playlist DidlPlaylistContainer that
+        matches the attribute specified.
+
+        Args:
+            attr_name (str): DidlPlaylistContainer attribute to compare. The
+                most useful being: 'title' and 'item_id'.
+            match (str): Value to match.
+
+        Returns:
+            (:class:`~.soco.data_structures.DidlPlaylistContainer`): The
+                first matching playlist object.
+
+        Raises:
+            (AttributeError): If indicated attribute name does not exist.
+            (ValueError): If a match can not be found.
+
+        Example::
+
+            obj.get_sonos_playlist_by_attr('item_id', 'SQ:3')
+            obj.get_sonos_playlist_by_attr('title', 'Foo')
+
+        """
+        for spl in self.get_sonos_playlists():
```

Comment:
In general, please write variable names out, especially if we are not short on space. attr for attribute is fine, because it is an established shorthand, but ``spl`` really should be ``sonos_playlist``. "Readability counts" ;)